src/odemis/driver/simcam.py

```diff
@@ -273,7 +275,8 @@ def _generate(self):
         if self._focus:
             # apply the defocus
             pos = self._focus.position.value['z']
-            dist = abs(pos - self._metadata[model.MD_FAV_POS_ACTIVE]["z"]) / (self.depthOfField.value * 10)
+            # max blur of sqrt(50) pixels, else image generation takes too long
+            dist = math.sqrt(min(abs(pos - self._metadata[model.MD_FAV_POS_ACTIVE]["z"]) / self.depthOfField.value, 50))
```

Comment:
I think that sqrt(50) is a little too small  now (7px). It's nice to be able to get a really blurry image. I'd suggest to make it more like `min(sqrt(...), 30)`.  In that case, it's mostly clipped by the sqrt, but still limited to 30 (for values > 900).